torch/_inductor/scheduler.py

```diff
@@ -4207,6 +4214,80 @@ def get_graph_partition_signature(
 
         return signatures[::-1]
 
+    def reorder_for_partition_based_on_devices(
+        self, nodes: list[BaseSchedulerNode]
+    ) -> list[BaseSchedulerNode]:
+        """
+        Given topologically sorted `nodes`, reorder by devices while
+        respecting dependencies. Specifically, when a node reads from
+        another device, execute all leading nodes on that device since
+        we already have a graph partition for that device.
+        """
+
+        result: list[BaseSchedulerNode] = []
+        device_to_nodes: dict[Optional[torch.device], list[BaseSchedulerNode]] = (
+            collections.defaultdict(list)
+        )
+
+        def visit(node: BaseSchedulerNode) -> None:
+            node_device = node.get_device()
+            for dep in node.unmet_dependencies:
+                if (
+                    buffer := self.name_to_buf.get(dep.name, None)
```

Comment:
when would name_to_buf.get(dep.name) not exist ? can we more explicitly handle the non case